tests/bwc/test_upgrade.py

```diff
@@ -343,3 +343,57 @@ def _move_nodes_folder_if_needed(self):
             target_path_nodes = os.path.join(self._path_data, 'nodes')
             shutil.move(src_path_nodes, target_path_nodes)
             shutil.rmtree(data_path_incl_cluster_id)
+
+class TableSettingsCompatibilityTest(NodeProvider, unittest.TestCase):
+
+    CLUSTER_SETTINGS = {
+        'cluster.name': gen_id(),
+    }
+
+    SUPPORTED_VERSIONS = (
+        '2.3.x',
+        'latest-nightly',
+    )
+
+    def test_altering_tables_with_old_settings(self):
+        nodes = 3
+
+        cluster = self._new_cluster(self.SUPPORTED_VERSIONS[0],
+                                    nodes,
+                                    self.CLUSTER_SETTINGS)
+        cluster.start()
+        with connect(cluster.node().http_url, error_trace=True) as conn:
+            cursor = conn.cursor()
+
+            # The used setting is only valid until version 2.3.x
+            cursor.execute('''
+                CREATE TABLE t1 (id int) with ("recovery.initial_shards"=1);
+            ''')
+            cursor.execute('''
+                CREATE TABLE p1 (id int, p int) partitioned by (p) with ("recovery.initial_shards"=1);
+            ''')
+            cursor.execute('''
+                INSERT INTO p1 (id, p) VALUES (1, 1);
+            ''')
+        self._process_on_stop()
+
+        for version in self.SUPPORTED_VERSIONS[1:]:
+            self.alter_tables(version, nodes)
+
+        # restart with latest version
+        self.alter_tables(self.SUPPORTED_VERSIONS[-1], nodes)
```

Comment:
why is this still needed? the previous call ( `self.SUPPORTED_VERSIONS[1:]`) starts a cluster on `latest-nightly` already

can we also remove `alter_table` to, maybe, `start_cluster_and_alter_number_of_replicas` or something similar (currently it's very generic and hides some of the stuff it's doing

Reply:
It is totally unrelated what setting is altered here, so I'm in big favour of keeping it like that. Seems like it is not completely clear what the test is doing, any suggestion to improve documentation wise?